File: Proyecto/ProyectoExploradorDeDirectorios/Compare.py
class Compare:

    alphabet="/0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"

    def __init__(self):
        pass
        
    def greaterLength(self,str1, str2):
        g = len(str1)
        if(len(str2) > g):
            g = len(str2)
       
        return g
            
    def lesserLength(self,str1, str2):
        l = len(str1)
        if(len(str2) < l):
            l = len(str2)
       
        return l
            
    def compare(self, object1, object2):
        
        if(isinstance(object1,int)):
            object1 = str(object1)

        if(isinstance(object1,str)):
            pass
        else:
            object1 = object1.name
        
        if(isinstance(object2,int)):
            object2 = str(object2)

        if(isinstance(object2,str)):
            pass
        else:
            object2 = object2.name

        lesser = self.lesserLength(object1,object2)
        for i in range(0,lesser,1):
            if(Compare.alphabet.index(object1[i]) > Compare.alphabet.index(object2[i])):
                return 1

            elif(Compare.alphabet.index(object1[i]) < Compare.alphabet.index(object2[i])):
                return -1
        if(lesser == self.greaterLength(object1,object2)):
            return 0

        elif(len(object1) == lesser):
            return -1

        else:
            return 1

File: 1pac/1.10-Arbol/Compare.js
function Compare(){

    alphabet = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";

    this.compare = CompareCompare;
    this.greaterLength = CompareGreaterLength;
    this.lesserLength = CompareLesserLength;

    
}


    function CompareGreaterLength(str1, str2){
        var g = str1.length;
        if(str2.length > g) g = str2.length;

        return g;
    }

    function CompareLesserLength(str1, str2){
        var l = str1.length;
        if(str2.length < l) l = str2.length;

        return l;
    }

    function CompareCompare(object1, object2){
        
        //object 1
        if(typeof object1 == "number"){
            num1 = object1
            object1 = `${object1}`.trim();
        }
        if(typeof object1 == "object")
            object1 = `${object1.name}`.trim();
            
        
        //object 2
        if(typeof object2 == "number"){
            num2 = object2
            object2 = `${object2}`.trim();
        }

        if(typeof object2 == "object")
            object2 = `${object2.name}`.trim();

        
        if(typeof num1 == "number" && typeof num2 == "number"){
            if(num1<num2)
                return -1
            else if(num1>num2)
                return 1
            else
                return 0
        }

        var lesser = this.lesserLength(object1,object2);
        for(var i = 0; i<lesser; i++){
            if(alphabet.indexOf(object1[i]) > alphabet.indexOf(object2[i]))
                return 1;

            else if(alphabet.indexOf(object1[i]) < alphabet.indexOf(object2[i]))
                return -1;
        
        }

        if(lesser == this.greaterLength(object1,object2))
            return 0;

        else if(object1.length == lesser)
            return -1;

        else
            return 1;
        
    }